core/include/gnuradio-4.0/ClaimStrategy.hpp

```diff
@@ -283,7 +284,16 @@ class alignas(hardware_constructive_interference_size) MultiThreadedStrategy
     }
 
 private:
-    void                      setAvailable(signed_index_type sequence) noexcept { setAvailableBufferValue(calculateIndex(sequence), calculateAvailabilityFlag(sequence)); }
+    void                      setAvailable(signed_index_type sequence) noexcept {
+        setAvailableBufferValue(calculateIndex(sequence), calculateAvailabilityFlag(sequence));
+        while (true) {
+            const auto oldReadCursor = _readCursor->value();
+            const auto newReadCursor = getHighestPublishedSequence(oldReadCursor + 1, _writeCursor.value());
```

Comment:
I think this is still not correct when actually using multiple writers, as getHighestPublishedSequence is operating on data shared between writers.